docs/source/faq_sby.rst

```diff
@@ -138,6 +138,90 @@ This includes:
+ever produce counterexamples where the loop is stable. For example, the following snippet
+creates a logic loop that is unstable only when the control signals ``(sx, sy)`` are both ``0``.
+
+.. code-block:: systemverilog
+
+   // (sx, sy) = (0, 0) gives an unstable logic loop
+   // (sx, sy) = (0, 1) gives (x, y) = (a + c + d, c + d    )
+   // (sx, sy) = (1, 0) gives (x, y) = (a + b    , a + b + c)
+   // (sx, sy) = (1, 1) gives (x, y) = (a + b    , c + d    )
+   assign x = a + (sx ? b : y);
+   assign y = c + (sy ? d : x);
+
+To break this loop for formal verification, one of the variables in the loop can be replaced with an
+``anyseq`` wire, and constrained with an assumption to take the desired value when it is known to
+be stable. It is also a good idea to add an assertion checking that the conditions leading to an
+unstable loop cannot happen. Note that even with this assertion, if unstable loops can occur in other
+cases the design could suffer from `overconstraint due to assumptions`_.
+
+.. code-block:: systemverilog
+
+   `ifdef FORMAL
+     // Define the conditions when we know the loop will be unstable. At all other
+     // times it is assumed to be stable which can hide counterexamples so care must be
+     // taken. We use an assertion to make sure the unstable condition can never be seen.
+     wire loop_unstable;
+     assign loop_unstable = {sx, sy} == '0;
+     always @* assert(!loop_unstable);
+
+     // Break the loop through x using an assumption when the loop is known to be stable
+     (* anyseq *) wire x;
+     always @* begin
+       if (!loop_unstable)
+         assume(x == a + (sx ? b : y)); 
+     end
+   `endif
+
+If the loop is introduced through a latch by multi-clock mode, sometimes the latch can be safely
+replaced with a flip-flop which doesn't have the same combinational paths. The standard clock-gating
+pattern shown below is an example of a circuit amenable to this technique.
+
+.. code-block:: systemverilog
+
+   module clock_gate(input wire clk_i, input wire en_i, output wire gated_clk_o);
+   // Latch means that en_l only changes when clk_i is low, so gated_clk_o cannot glitch
+   reg en_l;
+   always @* begin
+     if (!clk_i)
+       en_l = en_i;
+   end
+
+   assign gated_clk_o = en_l & clk_i;
+   endmodule
+
+This design rewritten with a flip-flop provides the same functionality for formal verification,
+but doesn't include the same combinational paths in multi-clock mode.
+
+.. code-block:: systemverilog
+
+   module clock_gate_formal(input wire clk_i, en_i, output wire gated_clk_o);
+   reg en_r;
+   always @(posedge clk_i)
+     en_r <= en_i;
+
+   assign gated_clk_o = en_r && clk_i;
+   endmodule
+
+Instances of ``clock_gate`` can be replaced with ``clock_gate_formal`` using
+``chtype -map clock_gate clock_gate_formal`` in the ``[script]`` section, and a separate formal
+testbench in multi-clock mode can be used to confirm equivalence of the two modules.
```

Comment:
I think this doesn't interact well with verific. It woud need very detailed descriptions of when it does/doesn't work going into the weeds of hierarchy resolution and the verific/rtlil boundary, so I don't think we want to recommend this. [Configurations](https://www.hdlworks.com/hdl_corner/verilog_ref/items/Configuration.htm) are supported with verific, I would recommend that instead. (We may need to add a FAQ entry about how to use configurations?)

Reply:
I think for ease for now I will just remove the reference to replacing those modules within Yosys as the question of how you go about replacing a module is largely orthogonal anyway